.github/workflows/CI.yml

```diff
@@ -93,7 +93,32 @@ jobs:
       with:
         name: binary_artifacts
         path: build
+        
+  analyze:
+    name: CodeQL Analyze
+    needs: build
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Initialize CodeQL
+        uses: github/codeql-action/init@v1
+        with:
+          languages: "go"
+        
+      - name: Download built artifacts
+        uses: actions/download-artifact@v2
+        with:
+          name: binary_artifacts
+          path: build
 
+      - name: Display structure of downloaded files
```

Comment:
maybe we can remove this step eventually if just for debugging purpose.

Reply:
Probably not needed. Currently it's just for debugging purposes. I'll leave it in for now until the CodeQL action is working properly for a bit.